.github/workflows/reusable_e2e_tests.yaml

```diff
@@ -0,0 +1,133 @@
+            cmake_opts: -DUSE_BUNDLED_DEPS=True
+          - name: asan
+            cmake_opts: -DUSE_BUNDLED_DEPS=True -DCMAKE_C_FLAGS=-fsanitize=address -DCMAKE_CXX_FLAGS=-fsanitize=address
+    runs-on: ubuntu-24.04
+    steps:
+      - name: Install deps ⛓️
+        run: |
+          sudo apt-get update && sudo apt-get install -y --no-install-recommends \
+            ca-certificates \
+            cmake \
+            build-essential \
+            git \
+            clang \
+            llvm \
+            pkg-config \
+            autoconf \
+            automake \
+            libtool \
+            libelf-dev \
+            wget \
+            libbpf-dev \
+            libcap-dev \
+            libtbb-dev \
+            libjq-dev \
+            libjsoncpp-dev \
+            libgtest-dev \
+            libprotobuf-dev \
+            protobuf-compiler \
+            "linux-headers-$(uname -r)"
+
+      - name: Checkout Libs ⤵️
+        uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4.2.2
+        with:
+          fetch-depth: 0
+
+      - name: Fix kernel mmap rnd bits
+        # Asan in llvm 14 provided in ubuntu 22.04 is incompatible with
+        # high-entropy ASLR in much newer kernels that GitHub runners are
+        # using leading to random crashes: https://reviews.llvm.org/D148280
+        run: sudo sysctl vm.mmap_rnd_bits=28
+
+      - name: Install deps ⛓️
+        run: |
+          sudo .github/install-deps.sh
+
+      - name: Build tests 🏗️
+        env:
+          # This avoids random failures on CI.
+          # (https://github.com/google/sanitizers/issues/1322#issuecomment-699946942)
+          ASAN_OPTIONS: intercept_tls_get_addr=0
+        run: |
+          mkdir -p build && cd build
+          cmake -DBUILD_BPF=ON \
+                -DBUILD_LIBSCAP_MODERN_BPF=ON \
+                -DBUILD_LIBSCAP_GVISOR=OFF \
+                -DENABLE_E2E_TESTS=ON \
+                ${{ matrix.cmake_opts }} \
+                -DUSE_BUNDLED_LIBBPF=ON \
+                ..
+          make -j$(nproc) sinsp-example driver bpf container_plugin
+          sudo -E make e2e-install-deps
+
+      - name: Use custom container plugin
+        if: inputs.custom_container_plugin != ''
+        run: |
+          mv ${{ inputs.custom_container_plugin }} build/test/e2e/container_plugin-prefix/src/container_plugin/
+
+      - name: Run tests - docker 🧪
+        env:
+          # This avoids random failures on CI.
+          # (https://github.com/google/sanitizers/issues/1322#issuecomment-699946942)
+          ASAN_OPTIONS: intercept_tls_get_addr=0
+        run: |
+          cd build
+          sudo -E ../test/e2e/scripts/run_tests.sh
+
+      - name: Archive test reports - docker
+        uses: actions/upload-artifact@ea165f8d65b6e75b540449e92b4886f43607fa02 # v4.6.2
+        if: failure()
+        with:
+          name: ${{ matrix.name }}_report
+          path: |
+            /tmp/report/
+
+      - name: Run tests - podman 🧪
```

Comment:
Wonder if the 2 `Run tests` steps should just be part of the build matrix 🤔 

They both look pretty similar with some additional configuration for podman that could be done on a separate step with a `if: container_engine == 'podman'` check.

Reply:
What we can do is taking 2 bool input parameters and then check them like: `if: inputs.docker` before executing tests.

Reply:
I mean, are we still aiming at cutting workflow lines? Because with your suggestion we would probably end up with a larger workflow :D 
Don't get me wrong, it's just that i don't see the point in making a state-of-the-art reusable workflow out of this :laughing: 
If you have a use case for it, feel free to share and i'll make the needed adjustments.

Reply:
> What we can do is taking 2 bool input parameters and then check them like: if: inputs.docker before executing tests, but i'd avoid increasing the test matrix cardinality.


I went with this solution in the end. Let me know if it works for you (we can touch it later if needed, but as i already said, i don't think we will ever touch this again :laughing: )

Reply:
> I mean, are we still aiming at cutting workflow lines? Because with your suggestion we would probably end up with a larger workflow :D

Well, we build the libs in quite a few places to be immediately tested and discarded, we could concentrate all builds on a first workflow, then do any testing needed on separate ones, but if that's too much work, who cares?


> it's just that i don't see the point in making a state-of-the-art reusable workflow out of this

I wouldn't call making builds more modular state-of-the-art, just good practice, but to each their own I guess.

Anyways, the changes are good enough I think.